weekly_report/generate_report.py

```diff
@@ -224,168 +228,227 @@ def get_website_names(self) -> list[str]:
-    # EPIC
-    individual_summary(c, epic, 3, "Epic")
-    c.save()
-
-
-def individual_summary(canvas_obj, website_stats_retriever: StatsRetriever, website_id: int, name: str):
-    """Creates individual report text for a website source."""
-    website_stats_retriever.tag_game_ratio()
-    website_stats_retriever.num_games_over_week()
-    canvas_obj.setFont('jersey_15', 30)
-    canvas_obj.setFillColorRGB(0.6, 0.09, 10.8)
-    canvas_obj.drawString(100, 750, f'GameScraper Weekly Report - {name}')
-    canvas_obj.drawImage('game_scraper_logo.png',
-                         x=520, y=735, width=50, height=50)
-
-    canvas_obj.setFont('jersey_15', 20)
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 680, 'Number of Games released:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    canvas_obj.drawString(
-        50, 660, f'{website_stats_retriever.total_num_games_released()}')
-
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 640, 'Average Price:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    canvas_obj.drawString(
-        50, 620, f'£ {website_stats_retriever.average_price()}')
-
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 580, 'Top 5 Tags:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    x = 560
-    for tag in website_stats_retriever.top_five_tags():
-        canvas_obj.drawString(50, x, f'- {tag}')
-        x -= 20
-
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 440, 'Top Platform:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    canvas_obj.drawString(50, 420, f'{website_stats_retriever.top_platform()}')
-
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 380, 'Average Rating:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    canvas_obj.drawString(
-        50, 360, f'{website_stats_retriever.average_rating()}')
-
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-
-    canvas_obj.drawImage(f'diagrams/tags_pie_chart_{website_id}.png',
-                         x=300, y=240, width=280, height=180)
-    x = 300
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 320, 'Top 3 Publishers:')
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    for publisher in website_stats_retriever.top_three_publishers():
-        canvas_obj.drawString(50, x, f'- {publisher}')
-        x -= 20
-    canvas_obj.drawImage(f'diagrams/chart_{website_id}.png', x=350, y=450)
-    x = 160
-    canvas_obj.setFillColorRGB(0.6, 0.50, 30.0)
-    canvas_obj.drawString(50, 200, 'Top 3 Developers:')
-    x = 180
-    canvas_obj.setFillColorRGB(0.6, 0.70, 35.0)
-    for developer in website_stats_retriever.top_three_developers():
-        canvas_obj.drawString(50, x, f'- {developer}')
-        x -= 20
-    canvas_obj.setFillColorRGB(0.6, 0.70, 30.0)
-    canvas_obj.drawString(300, 20, f'{canvas_obj.getPageNumber()}')
+class ReportMaker():
```

Comment:
Good use of OOP to manage and organise your functions!

Reply:
Yh really good